janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -937,8 +937,8 @@ private static void waitForRecoveriesToFinish(CloudSolrClient server, String col
                for (Map.Entry<String, Slice> entry : slices.entrySet()) {
                     Map<String, Replica> shards = entry.getValue().getReplicasMap();
                     for (Map.Entry<String, Replica> shard : shards.entrySet()) {
-                        String state = shard.getValue().getStr(ZkStateReader.STATE_PROP);
-                        if ((state.equals(Replica.State.RECOVERING) || state.equals(Replica.State.DOWN))
+                        String state = shard.getValue().getStr(ZkStateReader.STATE_PROP).toUpperCase();
+                        if ((state.equals(Replica.State.RECOVERING.name()) || state.equals(Replica.State.DOWN.name()))
```

Comment:
sorry i meant to do Replica.State.RECOVERING.equals(state) etc because you know RECOVERING is not null